Marlin/src/module/endstops.cpp

```diff
@@ -450,20 +534,24 @@ void Endstops::event_handler() {
        ENDSTOP_HIT_TEST_Z(),
       _ENDSTOP_HIT_TEST(I,'I'),
       _ENDSTOP_HIT_TEST(J,'J'),
-      _ENDSTOP_HIT_TEST(K,'K')
+      _ENDSTOP_HIT_TEST(K,'K'),
+      _ENDSTOP_HIT_TEST(M,'M'),
+      _ENDSTOP_HIT_TEST(O,'O'),
+      _ENDSTOP_HIT_TEST(P,'P'),
+      _ENDSTOP_HIT_TEST(Q,'Q')
     );
 
     #if USES_Z_MIN_PROBE_PIN
-      #define P_AXIS Z_AXIS
-      if (TEST(hit_state, Z_MIN_PROBE)) _ENDSTOP_HIT_ECHO(P, 'P');
+      #define S_AXIS Z_AXIS
+      if (TEST(hit_state, Z_MIN_PROBE)) _ENDSTOP_HIT_ECHO(S, 'S');
```

Comment:
'P' for Probe is sensible, and may be considered part of the protocol. Perhaps `TUVW` would be the better choice for axis names, rather than `MOPQ`. The `E` parameter is used in most instances to refer to the "Tool" so `T` should actually be non-conflicting in G-codes that take axis letters. Of course, with 10 axes the chances of parameter conflict is high.

Reply:
I would prefer to remove the 10th axis and go for `UVW`. I guess the confusion that may arise with configs that map the `I_AXIS` to letter U (`AXIS4_NAME 'U'`) would be acceptable.

Reply:
Thought about it once more. The name `U_AXIS` implies existance of secondary linear axes (usually parallel to X). This is nonsense for machines that have mainly rotational axes or that only have a linear axis parallel to z. Whenever possible, axis IDs `W` should stay reserved for references to a linear axis parallel to Z. Otherwise the implementation of the inverse kinematics for e.g. 7 axis robot arms or [for swiss style lathes with two independent z axes](https://cncmanual.com/download/2976/) will become very confusing. For now, I will remove the current P_AXIS and revert to `P` for probe.